[wdspec] `browsingContext.navigationCommitted` event

sadym-chromium opened the pull request.

sadym-chromium committed:
[wdspec] `browsingContext.navigationCommitted` event

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+PAGE_REDIRECTED_HTML = "/webdriver/tests/bidi/network/support/redirected.html"
+
+
+async def test_subscribe_unsubscribe_not_emitted(bidi_session):
```
```suggestion
async def test_unsubscribe(bidi_session):
```

OrKoN replied:
I think what it is testing is the unsubscribe method

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+        context=top_context["context"], url=test_page_nested_frames, wait="complete"
+    )
+
+    # Wait until we receive two events, one for the top context and one for
```
```suggestion
    # Wait until we receive three events, one for the top context and one for
```

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+    # Wait until we receive two events, one for the top context and one for
+    # each of the 2 iframes.
+    wait = AsyncPoll(bidi_session, timeout=2)
+    await wait.until(lambda _: len(events) == 3)
```
```suggestion
    await wait.until(lambda _: len(events) >= 3)
```

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+    remove_listener()
+
+
+async def test_same_document_navigation_not_emitted(bidi_session, new_tab, url, subscribe_events):
```
```suggestion
async def test_same_document(bidi_session, new_tab, url, subscribe_events):
```

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+
+
+@pytest.mark.parametrize("sandbox", [None, "sandbox_1"])
+async def test_document_write_not_emitted(bidi_session, subscribe_events, new_tab, sandbox):
```
```suggestion
async def test_document_write(bidi_session, subscribe_events, new_tab, sandbox):
```

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+    remove_listener()
+
+
+async def test_page_with_base_tag(
```
```suggestion
async def test_base_tag(
```

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+        await_promise=False,
+    )
+
+    # In the future we can wait for "browsingContext.contextCreated" event instead.
```
comment does not seem to be true

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+
+
+@pytest.mark.parametrize("type_hint", ["tab", "window"])
+async def test_new_context_not_emitted(bidi_session, subscribe_events, type_hint):
```
```suggestion
async def test_new_context(bidi_session, subscribe_events, type_hint):
```

OrKoN commented on webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py:
```diff
@@ -0,0 +1,518 @@
+
+
+@pytest.mark.parametrize("url", ["", "about:blank", "about:blank?test"])
+async def test_window_open_with_about_blank_not_emitted(
```
```suggestion
async def test_window_open_with_about_blank(
```

OrKoN reviewed (approved):
Generally, LGTM. Left a few comments. I think "not_emitted" suffixes can be dropped as the expectation is encoded in the test code for each scenario and might eventually change if the spec changes or an issue is discovered. Although I am not sure what is common across all bidi tests.

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
Update webdriver/tests/bidi/browsing_context/navigation_committed/navigation_committed.py

Co-authored-by: Alex Rudenko <OrKoN@users.noreply.github.com>

sadym-chromium committed:
comment

sadym-chromium commented:
> Generally, LGTM. Left a few comments. I think "not_emitted" suffixes can be dropped as the expectation is encoded in the test code for each scenario and might eventually change if the spec changes or an issue is discovered. Although I am not sure what is common across all bidi tests.

addressed your comments

sadym-chromium commented:
https://github.com/GoogleChromeLabs/chromium-bidi/issues/2982
